src/ipyradiant/visualization/interactive_exploration.py

```
@@ -0,0 +1,324 @@
+
+    @trt.observe("cyto_graph")
+    def update_cyto_graph(self, change):
+        self.cyto_graph.set_layout(name="cola")
+        self.cyto_graph.set_style(self.cyto_style)
+        # on is a callback for cyto_graph instance (must be set on each instance)
+        self.cyto_graph.on("node", "click", self.log_node_clicks)
+        # TODO: Why doesn't this update automatically?? Shouldn't it?
+        self.children = (
+            self.cyto_graph,
+            W.HBox(
+                children=[
+                    self.expand_button,
+                    self.undo_button,
+                    self.remove_temp_nodes_button,
+                ]
+            ),
+        )
+
+    @trt.validate("children")
+    def validate_children(self, proposal):
+        children = proposal.value
+        if not children:
+            children = (
+                self.cyto_graph,
+                W.HBox(
+                    children=[
+                        self.expand_button,
+                        self.undo_button,
+                        self.remove_temp_nodes_button,
+                    ]
+                ),
+            )
+        return children
+
+    def get_node(self, node):
+        """
+        This function is used to find a node given the id of a node copy. Used in the log_node_clicks
+        method to change the color of nodes.
+        """
+
+        for node_obj in self.cyto_graph.graph.nodes:
+            if node_obj.data["id"] == node["data"]["id"]:
+                return node_obj
+        # maybe return None and log warning?
+        raise ValueError("Node not found in cytoscape.graph.nodes.")
+
+    def log_node_clicks(self, node):
+        """
+        This function works with registering a click on a node. This will mark the node as selected and change the color of the
+        selected node.
+        """
+
+        try:
+            node_object = self.get_node(node)
+        except ValueError:
+            #     # logger.warn
```

TODO? 

I would actually just let this throw the error. If we click a node, and it can't find it, that should probably be a hard error.

---

So, just remove the try/except completely.